dmaws/rds.py

```diff
@@ -0,0 +1,206 @@
+import time
+
+import boto.rds
+import psycopg2
+import bcrypt
+
+from . import utils
+
+
+class RDS(object):
+    def __init__(self, region, logger=None, profile_name=None):
+        self.conn = boto.rds.connect_to_region(
+            region,
+            profile_name=profile_name)
+        self.log = logger or (lambda *args, **kwargs: None)
+
+    def get_instance(self, url=None, id=None):
+        for instance in self.conn.get_all_dbinstances():
+            if url and instance.endpoint[0] == url.split(':')[0]:
+                return instance
+            if id and instance.id == id:
+                return instance
+
+    def create_new_snapshot(self, snapshot_id, instance_id):
+        """Create a new RDS snapshot deleting the existing one if there
+        """
+        try:
+            self.conn.get_all_dbsnapshots(snapshot_id)
```

Comment:
Yes, this is how you get a specific db snapshot. If it doesn't exist it will raise a 404 short circuiting the delete.